### ISSUE TITLE
xcffib 0.10.1 breaks qtile 0.16.0+ 

### ISSUE THREAD

**Issue by knodalyte** [qtile/qtile#1839]
It appears that having too recent a version of xcffib causes major problems with qtile, in particular with multi-monitor setups. Initially the bar gets corrupted and a little later qtile crashes.

xcffib 0.10.1 is what gets installed by pip and/or Debian test. Downgrading to 0.8.1 fixed the problem.

A note in the changelog or install docs might be appropriate.

# Qtile version
0.16.1.dev2+g68455575

# Stack traces
2020-07-23 12:08:57,309 ERROR libqtile xcore.py:_xpoll():L278 Got an exception in poll loop
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/libqtile/backend/x11/xcore.py", line 231, in _xpoll
    event = self.conn.conn.poll_for_event()
  File "/home/cba/.local/lib/python3.8/site-packages/xcffib/__init__.py", line 571, in wrapper
    return f(*args)
  File "/home/cba/.local/lib/python3.8/site-packages/xcffib/__init__.py", line 612, in poll_for_event
    return self.hoist_event(e)
  File "/home/cba/.local/lib/python3.8/site-packages/xcffib/__init__.py", line 685, in hoist_event
    return self._process_error(ffi.cast("xcb_generic_error_t *", e))
  File "/home/cba/.local/lib/python3.8/site-packages/xcffib/__init__.py", line 648, in _process_error
    raise error(buf)
xcffib.xproto.MatchError

# Configuration

**Comment by tych0** [comment]
I've been running them together and they've been working just fine. There really isn't much difference on the user end between 0.8.1 and 0.10.1, so it seems like something else might be going on here.

Which xcb-proto are you built against, vs. what's on your system as libxcb?

**Comment by knodalyte** [comment]
libxcb is 1.14
xcb-proto is not installed

**Comment by tych0** [comment]
Ok, libxcb being 1.14 is good. Can you get an xtrace of it failing? Just wrapping your `exec qtile` in `exec xtrace qtile` should be enough.

**Comment by knodalyte** [comment]
I am unable to reproduce this error after pulling newer code (0.16.1.dev12+g9469e603). Qtile no longer becomes unusable sometimes after plugging in or removing a monitor, although I need to manually trigger a restart of qtile in order to restore it correctly. 

One thing that I discovered was that I my screen_change hook function was misconfigured (included a reference to qtile), apparently the newer code displayed an error that had been there for a long time:

`2020-07-25 11:45:53,314 ERROR libqtile hook.py:fire():L367 Error in hook screen_change
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/libqtile/hook.py", line 365, in fire
    i(*args, **kwargs)
TypeError: restart_on_randr() missing 1 required positional argument: 'ev'
`

And I upgraded again to xcffib 0.10.1 and everything still works.

So I think this is a non-issue and can be closed. Thank you and thanks all for qtile!
